File: src/1486-XOR-Operation-in-an-Array/xor_operation_in_an_array.js
/**
 * @param {number} n
 * @param {number} start
 * @return {number}
 */
var xorOperation = function(n, start) {
  let nums = []

  for (let i = 0; i < n; ++i) {
    nums.push(start + 2 * i)
  }
  
  let ans = nums[0]

  for (let i = 1; i < nums.length; ++i) {
    ans ^= nums[i]
  }
  
  return ans
};

let n = 4
let start = 3

console.log(xorOperation(n, start))

File: src/1486-XOR-Operation-in-an-Array/xor_operation_in_an_array.cpp
#include <iostream>
#include <vector>

class Solution {
public:
    int xorOperation(int n, int start) {
        std::vector<int> nums;

        for (int i = 0; i < n; ++i) {
            nums.push_back(start + 2 * i);
        }
        
        int ans = nums[0];

        for (size_t i = 1; i < nums.size(); ++i) {
            ans ^= nums[i];
        }
        
        return ans;
    }
};

int main() {
    int n = 4;
    int start = 3;
    Solution solution;

    std::cout << solution.xorOperation(n, start);

    return 0;
}